source/dub/dub.d

```diff
@@ -76,7 +77,12 @@ enum defaultRegistryURL = "http://code.dlang.org/";
 PackageSupplier[] defaultPackageSuppliers()
 {
 	logDiagnostic("Using dub registry url '%s'", defaultRegistryURL);
-	return [new RegistryPackageSupplier(URL(defaultRegistryURL))];
+	return [
+		new FallbackPackageSupplier(
+			new RegistryPackageSupplier(URL(defaultRegistryURL)),
+			new RegistryPackageSupplier(URL(fallbackRegistryURL))
+		)
+	];
```

Comment:
To avoid duplicates, e.g. in `dub search` and to correctly deal with user-facing error reporting. The current mechanism adds another registry which can be used e.g. to serve private projects, very different use-case than a mirror registry which is expected to serve identical projects.